server/datastore/mysql/maintained_apps_test.go

```diff
@@ -0,0 +1,87 @@
+package mysql
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	"github.com/fleetdm/fleet/v4/server/mdm/maintainedapps"
+	"github.com/go-kit/kit/log"
+	"github.com/jmoiron/sqlx"
+	"github.com/stretchr/testify/require"
+)
+
+func TestMaintainedApps(t *testing.T) {
+	ds := CreateMySQLDS(t)
+
+	cases := []struct {
+		name string
+		fn   func(t *testing.T, ds *Datastore)
+	}{
+		{"UpsertMaintainedApps", testUpsertMaintainedApps},
+		{"IngestWithBrew", testIngestWithBrew},
+	}
+
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			defer TruncateTables(t, ds)
+			c.fn(t, ds)
+		})
+	}
+}
+
+func testUpsertMaintainedApps(t *testing.T, ds *Datastore) {
+	ctx := context.Background()
+
+	listSavedApps := func() []*fleet.MaintainedApp {
+		var apps []*fleet.MaintainedApp
+		ExecAdhocSQL(t, ds, func(q sqlx.ExtContext) error {
+			return sqlx.SelectContext(ctx, q, &apps, "SELECT name, version, platform FROM fleet_library_apps ORDER BY token")
+		})
+		return apps
+	}
+
+	expectedApps := maintainedapps.IngestMaintainedApps(t, ds)
+	require.Equal(t, expectedApps, listSavedApps())
+
+	// ingesting again results in no changes
+	maintainedapps.IngestMaintainedApps(t, ds)
+	require.Equal(t, expectedApps, listSavedApps())
+
+	// upsert the figma app, changing the version
+	err := ds.UpsertMaintainedApp(ctx, &fleet.MaintainedApp{
+		Name:         "Figma",
+		Token:        "figma",
+		InstallerURL: "https://desktop.figma.com/mac-arm/Figma-999.9.9.zip",
+		Version:      "999.9.9",
+		Platform:     fleet.MacOSPlatform,
+	})
+	require.NoError(t, err)
+
+	// change the expected app data for figma
+	for _, app := range expectedApps {
+		if app.Name == "Figma" {
+			app.Version = "999.9.9"
+			break
+		}
+	}
+	require.Equal(t, expectedApps, listSavedApps())
+}
+
+func testIngestWithBrew(t *testing.T, ds *Datastore) {
+	if os.Getenv("NETWORK_TEST") == "" {
```

Comment:
This test calls the real brew API, so I put it behind that existing NETWORK_TEST env var. I believe this will still run on each CI, though. Are we ok doing this? It will generate a bit of load on the API, but on the other hand it's good to catch any unforeseen changes on their part.

Reply:
random thought: could we cache their responses in some other repo and then we only hit their API infrequently (once a day or something)?

Reply:
Maybe, I don't know ideally I'd like something easier, maybe we don't run that test in normal CI and only in our daily runs? I'll take a look if we already do this for some tests?

Reply:
hmm looks like we don't really do that at the moment... it could run only if the race detector is enabled, but that feels hacky... I don't know, I'll leave it like that for now but open to ideas/suggestions! Maybe it's not that bad to let it run too, given that we will hit it daily from all fleet installations...